spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -290,6 +311,9 @@ protected DatagramSocket getTheSocket() {
 	}
 
 	protected synchronized DatagramSocket getSocket() throws IOException {
+		if (this.socketExpression != null) {
+			return this.socketExpression.getValue(this.evaluationContext, DatagramSocket.class);
```

Comment:
This hasn't been addressed.
@yu55, have you faced some issues to fix the concern?
Just let us know and we will try to help you come up with the solution.